src/pages/RFOX/components/AddressSelection.tsx

```diff
@@ -1,39 +1,99 @@
 import {
+  Box,
   Button,
   Flex,
   FormControl,
   FormHelperText,
   FormLabel,
   Input,
-  Select,
   Stack,
 } from '@chakra-ui/react'
-import { type FC, useCallback, useMemo, useState } from 'react'
+import { fromAccountId, thorchainAssetId, thorchainChainId } from '@shapeshiftoss/caip'
+import type { FC } from 'react'
+import { useCallback, useMemo, useState } from 'react'
+import { useForm, useFormContext } from 'react-hook-form'
 import { useTranslate } from 'react-polyglot'
+import { AccountDropdown } from 'components/AccountDropdown/AccountDropdown'
+import type { TradeAmountInputFormValues } from 'components/MultiHopTrade/components/TradeAmountInput'
+import { validateAddress } from 'lib/address/address'
 
 type AddressSelectionProps = {
+  onRuneAddressChange: (address: string | undefined) => void
   isNewAddress?: boolean
 }
 
-export const AddressSelection: FC<AddressSelectionProps> = ({ isNewAddress }) => {
+const boxProps = {
+  width: 'full',
+}
+
+export type StakeValues = {
+  manualRuneAddress: string | undefined
+} & TradeAmountInputFormValues
+
+export const AddressSelection: FC<AddressSelectionProps> = ({
+  onRuneAddressChange,
```

Comment:
Should this be `handleRuneAddressChange`?

Reply:
Might as well! 4e5a3f62d6